polygeist/tools/cgeist/Lib/CGStmt.cc

```diff
@@ -997,6 +997,49 @@ ValueCategory MLIRScanner::VisitSwitchStmt(clang::SwitchStmt *stmt) {
   return nullptr;
 }
 
+ValueCategory MLIRScanner::VisitBinAssign(BinaryOperator *e) {
+  LLVM_DEBUG({
+    llvm::dbgs() << "VisitBinAssign: ";
+    e->dump();
+    llvm::dbgs() << "\n";
+  });
+  auto rhs = Visit(e->getRHS());
+  auto lhs = Visit(e->getLHS());
+
+  assert(lhs.isReference);
+  mlir::Value tostore = rhs.getValue(builder);
+  mlir::Type subType;
+  if (auto PT = lhs.val.getType().dyn_cast<mlir::LLVM::LLVMPointerType>())
+    subType = PT.getElementType();
+  else
+    subType = lhs.val.getType().cast<MemRefType>().getElementType();
+  if (tostore.getType() != subType) {
+    if (auto prevTy = tostore.getType().dyn_cast<mlir::IntegerType>()) {
```

Comment:
These code attempts to fix type mismatches for Integer types. Related to my question on lines 1009 and 1015, is it possible the LHS RHS have other types (floating point types, complex types, etc...).

Reply:
Good point. Will address in a different PR.

Reply:
Sure. I am OK with doing that in a separate PR. Same for my other comments.